backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/bpartner/bpartnercomposite/JsonRetrieverService.java

```
@@ -490,6 +494,31 @@ else if (ExternalIdentifier.Type.METASFRESH_ID.equals(bpartnerIdentifier.getType
 		return jsonMetasfreshId.map(metasfreshId -> MetasfreshId.of(metasfreshId.getValue()));
 	}
 
+	public Optional<BPartnerId> resolveBPartnerExternalIdentifier(@NonNull final ExternalIdentifier bPartnerExternalIdentifier, @NonNull final OrgId orgId)
+	{
+		switch (bPartnerExternalIdentifier.getType())
+		{
+			case METASFRESH_ID:
+				final BPartnerId bPartnerId = BPartnerId.ofRepoId(bPartnerExternalIdentifier.asMetasfreshId().getValue());
+				return Optional.of(bPartnerId);
+			case EXTERNAL_REFERENCE:
+				return externalReferenceService.getJsonMetasfreshIdFromExternalReference(orgId, bPartnerExternalIdentifier, BPartnerExternalReferenceType.BPARTNER)
+						.map(JsonMetasfreshId::getValue)
+						.map(BPartnerId::ofRepoId);
+			case GLN:
+				final BPartnerQuery bPartnerQuery = BPartnerQuery.builder()
+						.onlyOrgId(orgId)
+						.gln(bPartnerExternalIdentifier.asGLN())
+						.build();
+
+				return bpartnersRepo.retrieveBPartnerIdBy(bPartnerQuery);
+			default:
+				throw new AdempiereException("Given external identifier type is not supported!")
```

please throw an InvalidIdentifierException (subclass of AdempiereException)